src/base/projects/Blob.svelte

```diff
@@ -2,16 +2,17 @@
   import type { Blob } from "@app/project";
   import type { ProjectRoute } from "@app/router/definitions";
 
-  import Readme from "@app/base/projects/Readme.svelte";
   import HeaderToggleLabel from "@app/base/projects/HeaderToggleLabel.svelte";
   import ProjectLink from "@app/router/ProjectLink.svelte";
+  import Readme from "@app/base/projects/Readme.svelte";
   import { isMarkdownPath, scrollIntoView } from "@app/utils";
+  import { onMount } from "svelte";
   import { updateProjectRoute } from "@app/router";
 
+  export let activeRoute: ProjectRoute;
   export let blob: Blob;
-  export let line: string | null;
   export let getImage: (path: string) => Promise<Blob>;
-  export let activeRoute: ProjectRoute;
+  export let line: string | null;
```

Comment:
Can we align the types so we don't have both `null` and `undefined`?

```diff
diff --git a/src/base/projects/Blob.svelte b/src/base/projects/Blob.svelte
index 9f427f3..cd7709f 100644
--- a/src/base/projects/Blob.svelte
+++ b/src/base/projects/Blob.svelte
@@ -12,9 +12,9 @@
   export let activeRoute: ProjectRoute;
   export let blob: Blob;
   export let getImage: (path: string) => Promise<Blob>;
-  export let line: string | null;
+  export let line: string | undefined = undefined;

-  $: lineNumber = line ? parseInt(line.substring(1)) : null;
+  $: lineNumber = line ? parseInt(line.substring(1)) : undefined;

   const lastCommit = blob.info.lastCommit;
   const lines = blob.binary ? 0 : (blob.content.match(/\n/g) || []).length;
diff --git a/src/base/projects/Browser.svelte b/src/base/projects/Browser.svelte
index 43bd4db..f6b9509 100644
--- a/src/base/projects/Browser.svelte
+++ b/src/base/projects/Browser.svelte
@@ -28,7 +28,7 @@
   export let activeRoute: ProjectRoute;

   $: path = activeRoute.params.path || "/";
-  $: line = activeRoute.params.line || null;
+  $: line = activeRoute.params.line;

   // When the component is loaded the first time, the blob is yet to be loaded.
   let state: State = { status: Status.Loading, path };
```

Reply:
54871e3d